### ISSUE TITLE
[1.12.2] Max Hunger Value doesn't update visually & Saturation Text cut-off.

### ISSUE THREAD

**Issue by Z-Tunic** [yeelp/Scaling-Feast#33]
Minecraft: 1.12.2
Forge: 14.23.5.2847

Mods: 3
AppleCore-mc1.12.2-3.2.0
jei_1.12.2-4.15.0.291
scalingfeast-1.2.2-beta

=====

![](https://cdn.discordapp.com/attachments/377171551155650560/699298575359934494/scalingfeastmaxhungerbug.PNG)

It's a visual bug, as the actual max hunger still goes up but the number remains at 20 / x1. Also the Saturation Text has a slight cut-off.

**Comment by yeelp** [comment]
So I'm pretty sure this is literally caused by a single typo in code. Which also allows some other unintended behaviour. Commit ce7d485 addresses this, so that should be working nicely again.

The cutoff is a little more interesting. I've never seen it happen to me locally. I copied your overlay settings too. With the exact same saturation value, it looks fine.

![butitworksthoughidontgetitwhyyyyy](https://user-images.githubusercontent.com/22644157/79268918-8a966780-7e69-11ea-8cf8-b5db63610a28.png)

I'm not sure what I can do here. I'm rendering the saturation text as a single String, so there's nothing wrong with the placement of text characters on Scaling Feast's end. This could be caused by the GUI Scale options, Minecraft's Font Renderer, your specific monitor's resolution, or any combination of those 3.

Can you try Smooth Font and see if that makes the text render better? Looking at their CurseForge page, they mention that Minecraft has a hard time rendering text with fractional scaling (Which is exactly what's happening here). If I had to guess, I'd say that since the font renderer is trying to render such a small character at such a small scale, it might not render consistently across displays.

You can also try Scaling GUI's and see if that helps. You can fine tune the scaling of the HUD, which might help Minecraft just that little bit so that it allows it to draw the text correctly.

I'm going to close this issue, since I'm not sure there's much I can do with the cutoff in the text. Do let me know if any of the above solutions work.

**Comment by Z-Tunic** [comment]
Smooth Font will actually fix the cut off. The GUI scale I'm using is Normal so I dunno why it would cut-off unless it's the resolution I use which is 2560x1440 but it's windowed since I do a bit of background stuff while building or something so the res is slightly smaller.

**Comment by James103** [comment]
I have a 1920x1080 monitor, and Minecraft is running in windowed mode at 1920x1017. At a GUI scale of 'Small' or 'Medium', the saturation text is cut off as follows. A GUI scale of 'Large' or higher does not have this problem. I don't think it has to do with the text offset or any other settings, but I'll post the configs just in case.

![image](https://user-images.githubusercontent.com/37945304/82387394-89be9b80-99fc-11ea-96da-c5f33067ab9c.png)
[config.zip](https://github.com/yeelp/Scaling-Feast/files/4653750/config.zip)

**Comment by yeelp** [comment]
It's interesting that both of these cases cause problems with the saturation text and not the hunger text. Both texts are scaled the same way, the only difference is that the saturation text's x position is dependent on the length of the hunger text and _maybe_ that causes some floating point precision problems and _maybe_ that causes Minecraft's font renderer to mess things up? That's about the only thing I can think of. For a while, both on my laptop; where I started Scaling Feast's development and on my desktop; where I'm continuing Scaling Feast's development, I've been unable to replicate this bug. But this afternoon I had fiddled with the GUI settings and I managed to replicate it. Now I have the ability to troubleshoot this bug, and a few ideas that might actually fix it. So maybe this bug can be fixed after all.

I'm going to reopen this issue and take a look.
